packages/cloud_firestore/cloud_firestore/test/query_test.dart

```diff
@@ -202,6 +202,15 @@ void main() {
         );
       });
 
+      test(
+          'allow isNotEqualTo filter on FieldPath.documentId field & a different field on a separate filter',
+          () {
+        query!
+            .where(FieldPath.documentId, isNotEqualTo: 'fake-id')
+            .where(FieldPath.documentId, isEqualTo: 'another-fake-id')
+            .where('foo', isNull: true);
```

Comment:
The isNull case probably should be a separate test

Reply:
No, this test proves the logic works. We're trying to fix an issue with this query:
```dart
await FirebaseFirestore.instance
    .collection('users')
    .where(FieldPath.documentId, isNotEqualTo: 'FAKE_ID')
    .where(FieldPath.documentId, isEqualTo: 'USER_ID')
    .where('deletionDate', isNull: true)
    .limit(1)
    .get()
```

It's the combination of queries that matters, not in isolation.

Reply:
I see, alright then